axelrod/tests/strategies/test_memorytwo.py

```diff
@@ -1,11 +1,142 @@
 """Tests for the Memorytwo strategies."""
 
+import unittest
+import warnings
+
 import axelrod
 from .test_player import TestPlayer
-
+from axelrod.strategies.memorytwo import MemoryTwoPlayer
 
 C, D = axelrod.Action.C, axelrod.Action.D
 
+class TestGenericPlayerTwo(unittest.TestCase):
+    """A class to test the naming and classification of generic memory two
+    players."""
+    p1 = MemoryTwoPlayer(sixteen_vector=(0, 0, 0, 0, 0, 0, 0, 0,
+                                         0, 0, 0, 0, 0, 0, 0, 0))
+    p2 = MemoryTwoPlayer(sixteen_vector=(1, 0, 1, 0, 1, 0, 1, 0,
+                                         1, 0, 1, 0, 1, 0, 1, 0))
+
+    def test_name(self):
+        self.assertEqual(self.p1.name,
+                         "Generic Memory Two Player: (0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0)")
+        self.assertEqual(self.p2.name,
+                         "Generic Memory Two Player: (1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0)")
+
+    def test_stochastic_classification(self):
+        self.assertFalse(self.p1.classifier['stochastic'])
+        self.assertFalse(self.p2.classifier['stochastic'])
+
+class TestMemoryTwoPlayer(unittest.TestCase):
```

Comment:
@Nikoleta-v3 could we add some tests of the player with stochastic `sixteen_vector` (so using `TestPlayer.versus_test` with `init_kwargs`). Seeded tests, including same history/different seed to show that we indeed have stochastic behaviour?